components/calendar/src/date.rs

```diff
@@ -320,12 +320,30 @@ impl<A: AsCalendar> Date<A> {
         &self.calendar
     }
 
-    #[cfg(test)]
-    pub(crate) fn to_fixed(&self) -> calendrical_calculations::rata_die::RataDie {
+    #[doc(hidden)]
+    pub fn to_fixed(&self) -> calendrical_calculations::rata_die::RataDie {
         Iso::fixed_from_iso(self.to_iso().inner)
     }
 }
 
+impl Date<Iso> {
+    /// Calculates the number of days between two dates.
+    ///
+    /// ```rust
+    /// use icu::calendar::Date;
+    /// use icu::calendar::DateDurationUnit;
+    ///
+    /// let a = Date::try_new_iso(1994, 12, 10).unwrap();
+    /// let b = Date::try_new_iso(2024, 10, 30).unwrap();
+    ///
+    /// assert_eq!(b.days_since(a), 10_917);
+    /// ```
+    #[doc(hidden)] // unstable
```

Comment:
I think I would prefer embracing the replacement APIs but we can do that later

Reply:
There is already a hidden `until` API which is more powerful than this, so I don't want to stabilise this. `until` is not fully implemented, I'd have used it if the max-unit argument were respected.